### PR TITLE
LPS-198464 and LPS-197723 Update sites MVCResourceCommand with certain info + parameters and OOTB SXP Element features a Site Selector so the user can browse and select the sites to limit the search to without the need look-up for and copy-paste the groupIds (Site IDs)

### PR DESCRIPTION
**Pull request by felipelorenz**
Continued from here: https://github.com/brianchandotcom/liferay-portal/pull/145722
BChan request: https://github.com/brianchandotcom/liferay-portal/pull/145722#issuecomment-1870331518

### CODE REVIEW

**Comment by felipelorenz** [comment]
ci:test:sf

**Comment by felipelorenz** [comment]
ci:test:sf

**Comment by felipelorenz** [comment]
ci:test:sf

**Comment by felipelorenz** [comment]
ci:test:sf

**Comment by BryanEngler** [comment]
i havent looked at the code yet, but just as a note, i think we should probably try to avoid having commits from 2 different epics in the same pr...? in this case i think the code is probably related, but it makes it a harder on reviewers (and whoever might be testing it) when its like this

**Comment by BryanEngler** [comment]
https://github.com/felipelorenz/liferay-portal/compare/LPS-203746_test_MVC_and_check_rename_asked_by_BChan...BryanEngler:liferay-portal:2024_01_03_pr1208_liferay_search_LPS_198464_LPS_197723_sites_mvc_and_selector

@felipelorenz @gustavolimav what do you guys think about these changes?

**Comment by gustavolimav** [comment]
> [felipelorenz/liferay-portal@LPS-203746_test_MVC_and_check_rename_asked_by_BChan...BryanEngler:liferay-portal:2024_01_03_pr1208_liferay_search_LPS_198464_LPS_197723_sites_mvc_and_selector](https://github.com/felipelorenz/liferay-portal/compare/LPS-203746_test_MVC_and_check_rename_asked_by_BChan...BryanEngler:liferay-portal:2024_01_03_pr1208_liferay_search_LPS_198464_LPS_197723_sites_mvc_and_selector)
> 
> @felipelorenz @gustavolimav what do you guys think about these changes?

Looks good to me, probably it is what Bchan is expecting

**Comment by felipelorenz** [comment]
> [felipelorenz/liferay-portal@LPS-203746_test_MVC_and_check_rename_asked_by_BChan...BryanEngler:liferay-portal:2024_01_03_pr1208_liferay_search_LPS_198464_LPS_197723_sites_mvc_and_selector](https://github.com/felipelorenz/liferay-portal/compare/LPS-203746_test_MVC_and_check_rename_asked_by_BChan...BryanEngler:liferay-portal:2024_01_03_pr1208_liferay_search_LPS_198464_LPS_197723_sites_mvc_and_selector)
> 
> @felipelorenz @gustavolimav what do you guys think about these changes?

Looks good to me too!

**Comment by BryanEngler** [comment]
@felipelorenz can you cherry pick my commits, clean them up a little and then push them here please?

**Comment by felipelorenz** [comment]
> @felipelorenz can you cherry pick my commits, clean them up a little and then push them here please?

Sure!

**Comment by felipelorenz** [comment]
ci:test:sf

**Comment by felipelorenz** [comment]
ci:test:search

**Comment by BryanEngler** [comment]
@felipelorenz can you separate the changes in my commits into each module they touch, like we normally do with our commits. thanks!

**Comment by felipelorenz** [comment]
ci:test:sf

**Comment by felipelorenz** [comment]
ci:test:sf

File: modules/dxp/apps/search-experiences/search-experiences-web/src/main/java/com/liferay/search/experiences/web/internal/blueprint/admin/portlet/action/GetSitesMVCResourceCommand.java [hunk 1]
```diff
@@ -42,18 +40,8 @@ public boolean serveResource(
 		ResourceRequest resourceRequest, ResourceResponse resourceResponse) {
 
 		try {
-			String cmd = ParamUtil.getString(resourceRequest, Constants.CMD);
-
-			JSONObject jsonObject = null;
-
-			if (cmd.equals("getSiteJSONObject")) {
```
Comment (BryanEngler) [inline modules/dxp/apps/search-experiences/search-experiences-web/src/main/java/com/liferay/search/experiences/web/internal/blueprint/admin/portlet/action/GetSitesMVCResourceCommand.java]: @felipelorenz the original logic seems to imply that jsonObect should potentially be null (ie if cmd = "helloworld"), so just to keep things the same as they were before, i tihink we can probably add back the single 
`if (cmd.equals("getSiteJSONObject"))` check

Comment (BryanEngler) [reply modules/dxp/apps/search-experiences/search-experiences-web/src/main/java/com/liferay/search/experiences/web/internal/blueprint/admin/portlet/action/GetSitesMVCResourceCommand.java]: actually its "getSitesJSONObject" right?

Comment (felipelorenz) [reply modules/dxp/apps/search-experiences/search-experiences-web/src/main/java/com/liferay/search/experiences/web/internal/blueprint/admin/portlet/action/GetSitesMVCResourceCommand.java]: Yeah, you're right!

**Comment by BryanEngler** [comment]
ci:test:sf

**Comment by BryanEngler** [comment]
ci:test:relevant

**Comment by BryanEngler** [comment]
ci:test:search

**Comment by BryanEngler** [comment]
@felipelorenz i think the semver test failed on portal-search-api

**Comment by BryanEngler** [comment]
will forward from https://github.com/liferay-search/liferay-portal/pull/1215

**Pull request closed without merging**